pkg/interface/publish/src/js/lib/util.js

```diff
@@ -46,3 +46,26 @@ export function uxToHex(ux) {
   let value = ux.replace('.', '').padStart(6, '0');
   return value;
 }
+
+export function writeText(str) {
+  return new Promise(function (resolve, reject) {
+
+    var range = document.createRange();
+    range.selectNodeContents(document.body);
+    document.getSelection().addRange(range);
+
+    var success = false;
+    function listener(e) {
+      e.clipboardData.setData("text/plain", str);
+      e.preventDefault();
```

Comment:
Again, don't trust this due to race condition.

Reply:
This doesn’t materially change the race condition. All of the code inside of the copyListener Promise is still executed synchronously. The problem is that you want to be absolutely sure that “listener” has been called before returning from the promise.

Reply:
So, as said out of band, I'm sort of stuck on this; tried rewriting this with promises for different sections to make it a bit more ordered, and did some step-by-step debugging to see the control flow, which always seemingly just proceeds sequentially (isn't JS single-threaded anyway?) and in practice it works reliably like this, but we always find weird behaviours once we're in QA mode and on the network — however this is entirely client-side, too, so... I don't know here. Temporary pass?

Reply:
@loganallenc 's call. Can we just catch a possible crash so we don't bomb the whole page?

Reply:
I'm pretty sure it will just console.err. I can write a catch one sec